### PR TITLE
Revert "Revert "[eas-json] validate EAS Submit inputs better""

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Apply changes from https://github.com/expo/eas-cli/pull/2198 with fixes

# How

Same as for https://github.com/expo/eas-cli/pull/2198, but I made constraints for validating `ascApiKeyId` and `ascAppId` more loose and added some comments.

# Test plan

Tests

### CODE REVIEW

**Comment by szdziedzic** [comment]
/changelog-entry chore Add better validation for EAS Submit inputs

File: packages/eas-cli/src/submit/ios/IosSubmitCommand.ts [hunk 1]
```diff
@@ -111,6 +111,11 @@ export default class IosSubmitCommand {
     const envAppSpecificPassword = getenv.string('EXPO_APPLE_APP_SPECIFIC_PASSWORD', '');
 
     if (envAppSpecificPassword) {
+      if (!/^[a-z]{4}-[a-z]{4}-[a-z]{4}-[a-z]{4}$/.test(envAppSpecificPassword)) {
+        throw new Error(
+          'EXPO_APPLE_APP_SPECIFIC_PASSWORD must be in the format XXXX-XXXX-XXXX-XXXX, where X is a lowercase letter.'
```
Comment (wschurman) [inline packages/eas-cli/src/submit/ios/IosSubmitCommand.ts:116]:
```suggestion
          'EXPO_APPLE_APP_SPECIFIC_PASSWORD must be in the format xxxx-xxxx-xxxx-xxxx, where x is a lowercase letter.'
```

File: packages/eas-json/src/submit/schema.ts [hunk 2]
```diff
@@ -34,8 +36,46 @@ export const IosSubmitProfileSchema = Joi.object({
...
+  ascAppId: Joi.string()
+    .regex(/^\d+$/)
+    .message(
+      'Invalid Apple App Store Connect App ID ("ascAppId") was specified. It should consist only of digits. Example: "1234567891". Learn more: https://expo.fyi/asc-app-id.'
+    )
+    .max(30), // I didn't find any docs about it, but the longest app ID I've seen is 10 digits long so 30 characters limit should be enough
+  appleTeamId: Joi.string()
+    .regex(/^[\dA-Z]{10}$/) // Apple says that it always has to be 10 characters long https://developer.apple.com/help/account/manage-your-team/locate-your-team-id/
+    .message(
+      'Invalid Apple Team ID was specified. It should consist of 10 uppercase letters or digits. Example: "AB32CDE81F".'
```
Comment (wschurman) [inline packages/eas-json/src/submit/schema.ts:67]:
```suggestion
      'Invalid Apple Team ID was specified. It should consist of 10 uppercase letters or digits. Example: "AB32CZE81F".'
```
(so as not to confuse with hex, let's add some non-hex characters in the example)

**Review by wschurman — APPROVED** [review]

**Review by quinlanj — APPROVED** [review]
lgtm 🚀

**Pull request merged**